documentation/src/main/asciidoc/reference/coordination.asciidoc

```diff
@@ -0,0 +1,508 @@
++
+See <<mapper-orm-indexing-automatic-synchronization>> for details.
+[[coordination-database-polling-indexing-association-consistency]]
+Associations must be updated on both sides::
+Hibernate ORM is able to handle asymmetric updates of associations,
+where only the owning side of association is updated and the other side is ignored.
+The entities in the session will be inconsistent for the duration of the session,
+but upon reloading they will be consistent once again,
+due to how entity loading works.
++
+This dodgy practice of asymmetric updates of associations
+can cause problems in applications in general,
+but also in Hibernate Search specifically,
+where it may lead to out-of-sync indexes.
+Thus, it must be avoided.
++
+[NOTE]
+====
+For example, let's assume an indexed entity `A` has an `@IndexedEmbedded` association `A.b`to entity `B`,
+and that `B` owns that association on its side, `B.a`.
+One can just set `B.a` to `null` in order to remove the association between `A` and `B`,
+and the effect on the database will be exactly what we want.
+
+However, Hibernate Search will only be able to detect that `B.a` changed,
+and by the time it tries to infer which entities need to be reindexed,
+it will no longer be able to know what `B.a` used to refer to;
+that change in itself is useless to Hibernate Search,
+Hibernate Search will not know that `A` needs to be reindexed.
+It will "forget" to reindex `A`, leading to an out-of-sync index where `A.b` still contains `B`.
+
+In the end, the only way for Hibernate Search to know that `A` needs to be reindexed
+is to also set `A.b` to `null`, which will cause Hibernate Search to detect that `A.b` changed,
+and thus that `A` changed too.
+====
+
+[[coordination-database-polling-schema]]
+=== Impact on the database schema
+
+The database-polling coordination strategy needs to store data in additional tables in the application database,
+so that this data can be consumed by background threads.
+This includes in particular an outbox events table, where one row is pushed every time an entity is changed
+in a way that requires reindexing.
+
+These tables are accessed through entities that are automatically added to the Hibernate ORM configuration,
+and as such they should be automatically generated when relying on Hibernate ORM's
+link:{hibernateDocUrl}#configurations-hbmddl[automatic schema generation].
+
+If you need to integrate the creation/dropping of these tables to your own script,
+the easiest solution is to have Hibernate ORM generate DDL scripts for your whole schema
+and copy everything related to constructs (tables, sequences, ...) prefixed with `HSEARCH_`.
+See link:{hibernateDocUrl}#configurations-hbmddl[automatic schema generation],
+in particular the Hibernate ORM properties `javax.persistence.schema-generation.scripts.action`,
+`javax.persistence.schema-generation.scripts.create-target`
+and `javax.persistence.schema-generation.scripts.drop-target`.
```

Comment:
I know that would be difficult to maintain, but maybe we should provide the sql schemas for each databases.

Reply:
I really don't want to maintain that.